merlion/models/forecast/deep_base.py

```diff
@@ -0,0 +1,261 @@
+#
+# Copyright (c) 2022 salesforce.com, inc.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+#
+"""
+    Base class for Deep Learning Forecasting Models
+"""
+import copy
+import time
+
+import logging
+import numpy as np
+import pandas as pd
+
+
+from scipy.stats import norm
+from typing import List, Optional, Tuple, Union
+
+try:
+    import torch
+    import torch.nn as nn
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[deep-learning]` or "
+        "`pip install `salesforce-merlion[all]`"
+    )
+    raise ImportError(str(e) + ". " + err)
+
+
+from merlion.models.deep_base import DeepConfig, DeepModelBase
+from merlion.models.forecast.base import ForecasterBase, ForecasterConfig
+from merlion.models.utils.rolling_window_dataset import RollingWindowDataset
+from merlion.models.utils.time_features import get_time_features
+from merlion.models.utils.early_stopping import EarlyStopping
+
+from merlion.transform.base import TransformBase, Identity
+from merlion.transform.factory import TransformFactory
+from merlion.utils.misc import initializer, ProgressBar
+from merlion.utils.time_series import to_pd_datetime, to_timestamp, TimeSeries, AggregationPolicy, MissingValuePolicy
+
+
+logger = logging.getLogger(__name__)
+
+
+class DeepForecasterConfig(DeepConfig, ForecasterConfig):
+    """
+    Config object used to define a forecaster with deep model
+    """
+
+    def __init__(
+        self,
+        n_past: int,
+        start_token_len: int = 0,
```

Comment:
Can you remove `start_token_len` here and move it to the individual models which actually require it? I think this makes more sense especially since we've removed it from the dataset as well.